app/main.py

```diff
@@ -72,35 +72,44 @@ def _parse_sp_file(self):
             self.tenant_id = self._get_tenant_id(sp_data['tenantId'])
             self.client_id = sp_data['aadClientId']
             self.client_secret = sp_data['aadClientSecret']
+
             # in case use msi, potentially we could get mi client id from sp file as well
             if self.client_id == "msi" and self.client_secret == "msi":
                 self.user_assigned_identity_id = sp_data.get("userAssignedIdentityID", "")
-
+            
         _logger.info('Parsing Service Principle file completed')
 
     def _parse_sp_env(self):
         self.tenant_id = os.environ["TENANT_ID"]
         self.client_id = os.environ["CLIENT_ID"]
         self.client_secret = os.environ["CLIENT_SECRET"]
-        _logger.info('Parsing Service Principle env completed')
 
-    def _get_client(self):
+        _logger.info('Assign new azure identity env variable')
+        os.environ["AZURE_TENANT_ID"] = self.tenant_id
+        os.environ["AZURE_CLIENT_ID"] = self.client_id
+        os.environ["AZURE_CLIENT_SECRET"] = self.client_secret
+
+        _logger.info('Parsing Service Principle env completed')
+    
+    def _get_credential(self):
         if os.getenv("USE_MSI", "false").lower() == "true":
             _logger.info('Using MSI')
             if "MSI_CLIENT_ID" in os.environ:
                 msi_client_id = os.environ["MSI_CLIENT_ID"]
                 _logger.info('Using client_id: %s', msi_client_id)
-                credentials = MSIAuthentication(resource=VAULT_RESOURCE_NAME, client_id=msi_client_id)
+                credentials = ManagedIdentityCredential(client_id=msi_client_id)
             elif "MSI_OBJECT_ID" in os.environ:
                 msi_object_id = os.environ["MSI_OBJECT_ID"]
+                identity = {'object_id': msi_object_id}
                 _logger.info('Using object_id: %s', msi_object_id)
-                credentials = MSIAuthentication(resource=VAULT_RESOURCE_NAME, object_id=msi_object_id)
+                credentials = ManagedIdentityCredential(identity_config=identity)
             elif "MSI_RESOURCE_ID" in os.environ:
                 msi_resource_id = os.environ["MSI_RESOURCE_ID"]
+                identity = {'resource_id': msi_resource_id}
                 _logger.info('Using resource_id: %s', msi_resource_id)
-                credentials = MSIAuthentication(resource=VAULT_RESOURCE_NAME, msi_res_id=msi_resource_id)
+                credentials = ManagedIdentityCredential(identity_config=identity)
             else:
-                credentials = MSIAuthentication(resource=VAULT_RESOURCE_NAME)
+                credentials = DefaultAzureCredential()
```

Comment:
Is there a reason we don't use `ManagedIdentityCredential()` here? [DefaultAzureCredential](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) first tries to use a service principal configured by environment variables, and only after finding this option unavailable, tries to use a managed identity. So using `ManagedIdentityCredential`, if possible, should probably be slightly more performant.

Reply:
Good point, I think using ManagedIdentityCredential make more sense here.